core/arch/aarch64/instr_create.h

```diff
@@ -402,19 +405,6 @@
     opnd_create_reg_ex(opnd_get_reg(rm_or_imm), 0, DR_OPND_SHIFTED), \
     opnd_add_flags((sht), DR_OPND_IS_SHIFT), (sha)) : \
   instr_create_1dst_4src((dc), OP_add, (rd), (rn), (rm_or_imm), (sht), (sha))
-#define INSTR_CREATE_adds(dc, rd, rn, rm_or_imm) \
```

Comment:
Shouldn't this be kept?  What if an existing client is using it?  Parallel macros exist for _and, etc.

Reply:
We would like users to switch to the XINST_CREATE_add_s() macro  instead. Is that an acceptable inconvenience for users if it improves existing clients by using a platform independent macro?

Reply:
Encouraging the cross-platform is good, but it doesn't necessarily have to involve removing arch-specific capabilities.  

Why aren't `_sub` and `_and` treated the same way?

The XINST has dst-as-src: you could imagine a client writing aarch64-specific code and wanting 2 srcs separate from the dest.  I suppose they could use `_shift` with LSL,0, though the general idea is to have macros that try somewhat to match assembler mnemonics.

If you really don't want the base `_adds`, `_subs`, `_ands`, how about defining them as DO_NOT_USE_USE_XINST_CREATE_xxx_INSTEAD.  Though I would vote for keeping them just with a doxygen comment recommending the cross-platform instead.

Reply:
I see. Thanks for the explanation. I'll leave _adds in.